modules/dnn/src/dnn.cpp

```diff
@@ -154,6 +155,38 @@ Mat blobFromImages(const std::vector<Mat>& images, double scalefactor, bool swap
     return blob;
 }
 
+Mat blobFromImage(const Mat& image, const Size& size,
+                  const Scalar& mean, double scalefactor, bool swapRB)
+{
+    std::vector<Mat> images(1, image);
+    return blobFromImages(images, size, mean, scalefactor, swapRB);
+}
+
+Mat blobFromImages(const std::vector<Mat>& images_, Size size,
+                   const Scalar& mean_, double scalefactor, bool swapRB)
+{
+    std::vector<Mat> images = images_;
+    for (int i = 0; i < images.size(); i++)
+    {
+        Size imgSize = images[i].size();
+        if (size == Size())
+            size = imgSize;
+        float resizeFactor = std::max(size.width / (float)imgSize.width,
+                                      size.height / (float)imgSize.height);
+        images[i].convertTo(images[i], CV_32F);
+        resize(images[i], images[i], Size(), resizeFactor, resizeFactor);
+        Rect crop(Point(0.5 * (images[i].cols - size.width),
+                        0.5 * (images[i].rows - size.height)),
+                  size);
+        Scalar mean = mean_;
+        if (swapRB)
```

Comment:
It seems to me that with `swapRB == true` we'll do it twice. In this function and in origin `blobFromImages`.